src/runmode-unix-socket.c

```diff
@@ -831,6 +831,74 @@ TmEcode UnixSocketDatasetLookup(json_t *cmd, json_t *answer, void *data)
     }
 }
 
+static bool GetVlanParameters(json_t *cmd, json_t *answer, TrafficId *vlan_tuple)
+{
+    memset(vlan_tuple, 0, sizeof(*vlan_tuple));
+
+    TrafficId args;
+    uint16_t count = 0;
+
+    /* 3 get optional hargs (legacy)*/
+    json_t *hargs = json_object_get(cmd, "hargs");
+    if (hargs != NULL) {
+        if (!json_is_integer(hargs)) {
+            SCLogInfo("error: hargs not a number");
+            json_object_set_new(answer, "message", json_string("hargs not a number"));
+            return false;
+        }
+        if (json_integer_value(hargs) < 1 || json_integer_value(hargs) > 4094) {
+            SCLogInfo("error: vlan ids must be between 1 and 4094");
+            json_object_set_new(answer, "message", json_string("hargs invalid vlan id"));
+            return false;
+        } else {
+            args.vlan.tuple[count++] = (uint16_t)json_integer_value(hargs);
+        }
+    }
```

Comment:
We should return early, should we not ?
We are not expecting `hargs_tuple` if we got `hargs`